src/components/Simulation/index.js

```diff
@@ -21,29 +21,38 @@ const Transition = React.forwardRef(function Transition(props, ref) {
 });
 
 const mapPeriods = invoices => {
-  const all = invoices.map(invoice => ({
-    key: invoice.period,
-    human: humanize(invoice.period),
-  }));
-  return uniqWith(all, (a, b) => a.key === b.key);
+  if (invoices) {
+    const all = invoices.map(set => ({
+      key: set.period,
+      human: humanize(set.period),
+    }));
+    return uniqWith(all, (a, b) => a.key === b.key);
+  }
+  return [];
 };
 
 const mapPackages = invoices => {
-  const all = invoices.map(invoice => ({
-    key: invoice.code,
-    human: humanize(invoice.code),
-  }));
-
-  return uniqWith(all, (a, b) => a.key === b.key);
+  if (invoices) {
```

Comment:
`(invoice || [])` would that come at a performance cost? Because I kind of like not having to thing about the `if`s

Reply:
imho, it’s only moving the check somewhere else. I hesitated to just use some let vars then only call the mapping function if invoices are defined.

Reply:
The check moves but you only have one return value to follow, that's the biggest benefit imho.